zipkin-web/src/main/resources/app/js/component_ui/timeStamp.js

```diff
@@ -28,7 +28,9 @@ define(
       }
 
       this.dateChanged = function (e) {
-        this.setTimestamp(moment(e.date));
+        var time = moment(e.date);
+        time.add(moment.duration(this.$time.val()));
```

Comment:
I found that `e.date` contains date only, no time.
So before this change, if you changed date, time will be cleared to 00:00